product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -29,6 +29,8 @@ def price_compute(
         if price_type == 'partner':
             partner = self.env.context.get('partner_id', False) or \
                 self.env.context.get('partner', False)
+            if partner and isinstance(partner, models.BaseModel):
+                partner = partner.id
```

Comment:
But if this is an ID, why the variable is called `partner`? It should be `partner_id` (I know that the code may not be yours, but we have now the opportunity to fix this).

Reply:
done